test/unit/auth/tenant.spec.ts

```diff
@@ -0,0 +1,272 @@
+            .to.throw(`"CreateTenantRequest.type" must be either "full_service" or "lightweight".`);
+        });
+      });
+
+      it('should throw on invalid EmailSignInConfig', () => {
+        const tenantOptionsClientRequest: TenantOptions = deepCopy(clientRequest);
+        tenantOptionsClientRequest.emailSignInConfig = null;
+        tenantOptionsClientRequest.type = 'full_service';
+
+        expect(() => Tenant.buildServerRequest(tenantOptionsClientRequest, true))
+          .to.throw('"EmailSignInConfig" must be a non-null object.');
+      });
+
+      const nonObjects = [null, NaN, 0, 1, true, false, '', 'a', [], [1, 'a'], _.noop];
+      nonObjects.forEach((request) => {
+        it('should throw on invalid CreateTenantRequest:' + JSON.stringify(request), () => {
+          expect(() => {
+            Tenant.buildServerRequest(request as any, true);
+          }).to.throw('"CreateTenantRequest" must be a valid non-null object.');
+        });
+      });
+
+      it('should throw on unsupported attribute for create request', () => {
+        const tenantOptionsClientRequest = deepCopy(clientRequest) as any;
+        tenantOptionsClientRequest.unsupported = 'value';
+        expect(() => {
+          Tenant.buildServerRequest(tenantOptionsClientRequest, true);
+        }).to.throw(`"unsupported" is not a valid CreateTenantRequest parameter.`);
+      });
+
+      const invalidTenantNames = [null, NaN, 0, 1, true, false, '', [], [1, 'a'], {}, { a: 1 }, _.noop];
+      invalidTenantNames.forEach((displayName) => {
+        it('should throw on invalid CreateTenantRequest displayName:' + JSON.stringify(displayName), () => {
+          const tenantOptionsClientRequest = deepCopy(clientRequest) as any;
+          tenantOptionsClientRequest.displayName = displayName;
+          expect(() => {
+            Tenant.buildServerRequest(tenantOptionsClientRequest, true);
+          }).to.throw('"CreateTenantRequest.displayName" must be a valid non-empty string.');
+        });
+      });
+
+      invalidTypes.forEach((invalidType) => {
+        it('should throw on creation with invalid type ' + JSON.stringify(invalidType), () => {
+          const tenantOptionsClientRequest: TenantOptions = deepCopy(tenantOptions);
+          tenantOptionsClientRequest.type = invalidType as any;
+          expect(() => Tenant.buildServerRequest(tenantOptionsClientRequest, true))
+            .to.throw(`"CreateTenantRequest.type" must be either "full_service" or "lightweight".`);
+        });
+      });
+    });
+  });
+
+  describe('getTenantIdFromResourceName()', () => {
+    it('should return the expected tenant ID from resource name', () => {
+      expect(Tenant.getTenantIdFromResourceName('projects/project1/tenants/TENANT_ID'))
+        .to.equal('TENANT_ID');
+    });
+
+    it('should return the expected tenant ID from resource name with multiple tenants substring', () => {
```

Comment:
The only difference from the previous test case is that this seems to use a different project ID. Was that intentional?

Reply:
I updated the description. I am testing when the projectId contains the "tenants" substring.